src/exposurelog/routers/add_message.py

```diff
@@ -85,11 +84,11 @@ async def add_message(
     el_table = state.exposurelog_db.table
 
     # Add the message.
-    async with state.exposurelog_db.engine.acquire() as connection:
-        result_proxy = await connection.execute(
+    async with state.exposurelog_db.engine.begin() as connection:
+        result = await connection.execute(
             el_table.insert()
             .values(
-                date_added=date_added,
+                date_added=curr_tai.tai.datetime,
```

Comment:
Yes converting TAI to datetime works. It is safer than converting UTC because TAI cannot have leap seconds.

Reply:
Regarding fractional seconds: do you have a suggestion? I looked for alternatives and didn't find any. PostgreSQL documents DateTime as having microsecond precision https://www.postgresql.org/docs/9.1/datatype-datetime.html, which may be enough to assure this will work.

I did look for alternatives but I find the doc confusing. I found a TIMESTAMP field for MySQL but it has the same rules as DATETIME except it MySQL helpfully assumes it is in the local timezone and converts it to UTC, which I don't want.

Reply:
Date/time types are indeed very backend-specific and their behavior is not very consistent. If you want to limit yourself to postgres then DateTime might work OK for you. In other places when portability between databases is important people frequently ignore native database time types and use integers with reasonable precision (micro-, nanoseconds) to keep offsets since some epoch time. That is not always convenient to work with but give a lot of control over all time issues (timezones, scales, etc,)